pkg/listen.go

```diff
@@ -14,6 +14,12 @@ import (
 	"github.com/tantosec/oneshell/pkg/patching"
 )
 
+type Listener struct {
+	Listen  func(n string, addr string) (net.Listener, error)
+	Address string
+	Port    uint16
+}
+
 func resolveIP(target string) (net.IP, error) {
```

Comment:
This is a good point, however I'm not sure if I want to support IPv6 yet as it would require a fair amount of work changing the assembly code to be able to handle it. I might raise a Github issue for it